### PR TITLE
feat(batch): Introducing local execution mode

### PR DESCRIPTION
**Pull request by liurenjie1024**
## What's changed and what's your intention?

Initial introduction of local execution mode. Currently we can't complex query with table scan. Which will be added later.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]
This PR looks pretty good to me.

File: src/frontend/Cargo.toml [hunk 1]
```diff
@@ -29,12 +29,14 @@ parking_lot = "0.12"
...
 pgwire = { path = "../utils/pgwire" }
 rand = "0.8"
+risingwave_batch = { path = "../batch" }
 risingwave_common = { path = "../common" }
 risingwave_expr = { path = "../expr" }
 risingwave_pb = { path = "../prost" }
 risingwave_rpc_client = { path = "../rpc_client" }
 risingwave_source = { path = "../source" }
 risingwave_sqlparser = { path = "../sqlparser" }
+risingwave_storage = { path = "../storage" }
```
Comment (fuyufjh) [inline src/frontend/Cargo.toml:39]: Why frontend depends on storage and batch?

**Review by fuyufjh — CHANGES_REQUESTED** [review]

File: src/frontend/Cargo.toml (hunk 1 above)
Comment (liurenjie1024) [reply src/frontend/Cargo.toml:39]: Because the task context trait requires it.

Comment (liurenjie1024) [reply src/frontend/Cargo.toml:39]: It depends on batch crate anyway, but we can do refactor to remove dependency to storage crate. Let's   merge this first, and I'll do refactor later to remove dependency to storage. cc @fuyufjh

**Review by fuyufjh — APPROVED** [review]

File: src/frontend/Cargo.toml (hunk 1 above)
Comment (liurenjie1024) [reply src/frontend/Cargo.toml:39]: Sure, see #2743

**Pull request merged**